### PR TITLE
fix(uninstall): ignore not found error when deleting CR

### PR DESCRIPTION
**Pull request by ChanYiLin**
ref: https://github.com/longhorn/longhorn/issues/9303

We should ignore not found error when deleting resources

### CODE REVIEW

File: controller/uninstall_controller.go [hunk 1]
```diff
@@ -647,13 +647,13 @@ func (c *UninstallController) deleteVolumes(vols map[string]*longhorn.Volume) (e
 
 		timeout := metav1.NewTime(time.Now().Add(-gracePeriod))
 		if vol.DeletionTimestamp == nil {
-			if err = c.ds.DeleteVolume(vol.Name); err != nil {
+			if err = c.ds.DeleteVolume(vol.Name); err != nil && !datastore.ErrorIsNotFound(err) {
 				err = errors.Wrap(err, "failed to mark for deletion")
 				return
 			}
 			log.Info("Marked for deletion")
```
Comment (derekbit) [inline controller/uninstall_controller.go:654]: Suggest 
```
			if err = c.ds.DeleteVolume(vol.Name); err != nil {
				if datastore.ErrorIsNotFound(err) {
					log.Info("Volume is not found")
				} else {
					err = errors.Wrap(err, "failed to mark for deletion")
					return
				}
			} else {
				log.Info("Marked for deletion")
			}
```

Comment (ChanYiLin) [reply controller/uninstall_controller.go]: fixed

**Review by derekbit — APPROVED** [review]
LGTM

**Comment by derekbit** [comment]
@mergify backport v1.7.x

**Pull request merged**